File: 2220. Minimum Bit Flips to Convert Number/minBitFlips.js
/**
 * @param {number} start
 * @param {number} goal
 * @return {number}
 */
const minBitFlips = (start, goal) => {
  let flipCount = 0;

  while (start > 0 || goal > 0) {
    const startBit = start & 1;
    const goalBit = goal & 1;

    if (startBit !== goalBit) {
      flipCount++;
    }

    start >>= 1;
    goal >>= 1;
  }

  return flipCount;
};


const start = 10;
const goal = 7;
console.log("minBitFlips: ", minBitFlips(start, goal)); // 2

File: 2220. Minimum Bit Flips to Convert Number/minBitFlips.py
class Solution:
  def minBitFlips(self, start: int, goal: int) -> int:
    flipCount = 0

    while start > 0 or goal > 0:
      startBit = start & 1
      goalBit = goal & 1

      if startBit != goalBit:
        flipCount += 1

      start >>= 1
      goal >>= 1

    return flipCount


start = 10
goal = 7
solution = Solution()
print("minBitFlips: ", solution.minBitFlips(start, goal))
